File: src/manager/Models/Account.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace easyTradeManager.Models
{
    public class Account
    {
        public int Id { get; set; }
        public int PackageId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Username { get; set; }
        public string Email { get; set; }
        public string HashedPassword { get; set; }
        public string Origin { get; set; }
        public DateTimeOffset CreationDate { get; set; }
        public DateTimeOffset PackageActivationDate { get; set; }
        public Boolean AccountActive { get; set; }
        public string Address { get; set; }
    }
}

File: src/db-adapter/internal/models/account.go
package models

import (
	"context"
	"time"
)

type Account struct {
	ID                    string
	PackageID             string
	FirstName             string
	LastName              string
	Username              string
	Email                 string
	HashedPassword        string
	Origin                string
	CreationDate          time.Time
	PackageActivationDate time.Time
	AccountActive         bool
	Address               string
}

type AccountRepository interface {
	Create(ctx context.Context, account *Account) (*Account, error)
	GetByID(ctx context.Context, id string) (*Account, error)
	GetByUsername(ctx context.Context, username string) (*Account, error)
	GetAll(ctx context.Context) ([]*Account, error)
	DeleteOlderThan(ctx context.Context, date time.Time, origin string) (int32, error)
}
